rfcs/090-lti1.3-implmentation-needs.md

```diff
@@ -0,0 +1,110 @@
+# Summary
+
+This is a summary of changes that need to happen to add LTI 1.3 support into PrairieLearn
+to support integration with learning management systems like Canvas. Notes may be collected
+here or broken out to their own RFCs.
+
+This guide could be broken down into a versioned timeline, for what is the
+bare minimum needed for Canvas integration versus more fully embracing running
+PL in an embedded app environment (for both students and instructors).
+
+References:
+* https://www.imsglobal.org/spec/lti/v1p3/
+
+# Table of Contents
+* LTI configuration endpoints
+* LTI OAuth 2.0 JSON Web Token authentication
+* Authentication session tracking
+* PrairieLearn in an IFRAME
+* Pages to support boostrapping from Canvas
+* Decide what to do with pre-1.3 LTI support
+
+
+# LTI configuration endpoints
+
+Canvas can be configured for an LTI app (their term) by pointing to an JSON URL that returns
+the configurations parameters needed. This is preferred because it makes setup on the LMS easy.
+
+That may also include public key targets or pieces needed for the JWT authentication and
+validations.
+
+# LTI OAuth 2.0 JSON Web Token authentication
+
+We need an authCallback and authLogin method for OAuth 2.0 authentications from LTI 1.3.
+
+https://www.imsglobal.org/spec/security/v1p0
+
+This authentication target could be sourced from a library or homegrown. It's not
+(necessarily) exchangable with the current authLoginOAuth2.js that uses Google.
+
+* Rename the existing authLoginOAuth2 to authLoginGoogle
+
+# Authentication session tracking
+
+(This will probably be broken out to its own RFC since it's scoped larger than LTI.)
+
+One of the things we would like to overcome with the current LTI implementation is that
+it is conceivable that users could connect to PrairieLearn directly via one auth source
+(i.e. connecting to https://prairielearn.org and authenticating with Shibboleth) while
+also being logged in via LTI (i.e. from Canvas linked into a course). Currently, these
+authentications fight with each other as only one can be active at a time.
+
+We should think about how to track multiple authorizations to a single session.
+
+Server side session tracking would also give us the ability to invalidate current sessions, which is desirable.
+* The server side session feature is being tracked in issue #5416.
+
+Session tracking would also allow us to cache and implement auth renewals as needed,
+potentially seamless to the user.
+
+# PrairieLearn in an IFRAME
```

Comment:
This section is still relevant. One of my development branches adds a navbar when it detects PL in an IFRAME, which is helpful for development. There are things to unpack here, but are bigger than LTI.